iceoryx_hoofs/include/iceoryx_hoofs/cxx/newtype.hpp

```diff
@@ -119,6 +122,40 @@ class NewType : public Policies<NewType<T, Policies...>>...
   private:
     T m_value;
 };
+
+/// @brief This macro helps to create types with the NewType class.
+/// In case the functionality of the underlying type is sufficient and one just
+/// wants to prevent to mix types, this macro removes the burden to write
+/// boilerplate code.
+///
+/// @param[in] TypeName is the name of the new type to create
+/// @param[in] Type is the underlying type of the new type
+/// @param[in] ... is a variadic list of the policies applied to the new type,
+/// e.g. iox::cxx::newtype::ConstructByValueCopy
+///
+/// @code
+///     #include <iceoryx_hoofs/cxx/newtype.hpp>
+///
+///     IOX_NEW_TYPE(MyType,
+///                  uint64_t,
+///                  newtype::ConstructByValueCopy,
+///                  newtype::Comparable,
+///                  newtype::Sortable,
+///                  newtype::AssignByValueCopy)
+/// @endcode
+#define IOX_NEW_TYPE(TypeName, Type, ...)                                                                              \
+    struct TypeName : public iox::cxx::NewType<Type, __VA_ARGS__>                                                      \
+    {                                                                                                                  \
+        using ThisType::ThisType;                                                                                      \
+        using ThisType::operator=;                                                                                     \
+                                                                                                                       \
+        TypeName() noexcept = default;                                                                                 \
+        TypeName(const TypeName&) noexcept = default;                                                                  \
+        TypeName(TypeName&&) noexcept = default;                                                                       \
+        TypeName& operator=(const TypeName&) noexcept = default;                                                       \
+        TypeName& operator=(TypeName&&) noexcept = default;                                                            \
+    };
```

Comment:
The big question, semicolon or not. Without the semicolon the macro usage would look like this
```cpp
IOX_NEW_TYPE(Foo, uint64_t, newtype::ConstructByValueCopy, newtype::Comparable);
```
It is more familiar to the manual creation via the struct and the error message is ` error: expected ';' after struct`, which is also clear. Therefore I would remove the semicolon from the macro. What do you think